stock_no_negative/tests/test_stock_no_negative.py

```diff
@@ -166,16 +166,17 @@ def test_true_allow_negative_stock_product_with_lot(self):
         the allow_negative_stock is set active in the product with lot"""
         self.product_with_lot.allow_negative_stock = True
         self.stock_picking_with_lot.action_confirm()
-        self.stock_picking_with_lot.move_ids.quantity_done = 100
+        self.stock_picking_with_lot.move_ids.quantity = 100
         with self.assertRaises(UserError):
+            self.stock_picking_with_lot.move_ids.picked = True
             self.stock_picking_with_lot._action_done()
         self.stock_picking_with_lot.move_ids.move_line_ids[0].lot_id = self.lot1.id
         self.stock_picking_with_lot._action_done()
+        self.stock_picking_with_lot.button_validate()
```

Comment:
Because you call  `button_validate` function after `_action_done` I suspect the test is failing due to another cause, and this is only hiding the problem.